homeassistant/components/isy994/config_flow.py

```diff
@@ -173,14 +216,21 @@ async def async_step_ssdp(self, discovery_info):
         """Handle a discovered isy994."""
         friendly_name = discovery_info[ssdp.ATTR_UPNP_FRIENDLY_NAME]
         url = discovery_info[ssdp.ATTR_SSDP_LOCATION]
+        parsed_url = urlparse(url)
         mac = discovery_info[ssdp.ATTR_UPNP_UDN]
         if mac.startswith(UDN_UUID_PREFIX):
             mac = mac[len(UDN_UUID_PREFIX) :]
         if url.endswith(ISY_URL_POSTFIX):
             url = url[: -len(ISY_URL_POSTFIX)]
 
-        await self.async_set_unique_id(mac)
-        self._abort_if_unique_id_configured()
+        if parsed_url.port:
+            port = parsed_url.port
+        elif parsed_url.scheme == SCHEME_HTTPS:
+            port = HTTPS_PORT
+        else:
+            port = HTTP_PORT
```

Comment:
Styling, I just don't like "else"... 😉 

```suggestion
        port = HTTP_PORT
        if parsed_url.port:
            port = parsed_url.port
        elif parsed_url.scheme == SCHEME_HTTPS:
            port = HTTPS_PORT
```

Reply:
Works for me